src/keystore/InMemoryKeystore.ts

```diff
@@ -254,6 +265,41 @@ export default class InMemoryKeystore implements Keystore {
     }
   }
 
+  async signDigest(
+    req: keystore.SignDigestRequest
+  ): Promise<signature.Signature> {
+    if (!validateObject(req, ['digest'], [])) {
+      throw new KeystoreError(
+        ErrorCode.ERROR_CODE_INVALID_INPUT,
+        'missing required field'
+      )
+    }
+
+    const { digest, identityKey, prekeyIndex } = req
+    let key: PrivateKey
+    if (identityKey) {
+      key = this.v1Keys.identityKey
+    } else if (
+      typeof prekeyIndex !== 'undefined' &&
```

Comment:
Out of curiosity, is there a case where signing would ever occur using not the most currentPreKey? Does validation of signatures use with path? as well?

Reply:
The previous behaviour was to always sign with whatever preKey was at the `0` index (newest). Validation of signatures is done against whatever `preKey` is included in the message as part of the `sender` field.